src/Analyzers/CSharp/CodeFixes/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -177,7 +180,21 @@ private static bool IsIEnumerator(ITypeSymbol returnType, KnownTaskTypes knownTy
     private static (SyntaxTokenList newModifiers, TypeSyntax newReturnType) AddAsyncModifierWithCorrectedTrivia(SyntaxTokenList modifiers, TypeSyntax returnType)
     {
         if (modifiers.Any())
-            return (modifiers.Add(s_asyncKeywordWithSpace), returnType);
+        {
+            // 'partial' modifier must say at the end of the modifiers arrays.
+            var partialModifier = modifiers.FirstOrDefault(static m => m.IsKind(SyntaxKind.PartialKeyword));
```

Comment:
>FirstOrDefault [](http://example.com/codeflow?start=44&length=14)

isn't there an indexof that takes in a kind to find?

Reply:
Nope!

Reply:
What about [this](https://github.com/dotnet/roslyn/blob/2d7ee0d5da8eebaf39e6d01fb6be6f67c3a64d55/src/Compilers/Core/Portable/Syntax/SyntaxTokenList.cs#L297), it's internal, but I thought it would be accessible? Works the way you have it, and I know the lists are small, but it seems weird to always search to find it.

Reply:
> What about [this](https://github.com/dotnet/roslyn/blob/2d7ee0d5da8eebaf39e6d01fb6be6f67c3a64d55/src/Compilers/Core/Portable/Syntax/SyntaxTokenList.cs#L297), it's internal,

Nope.  That's in compiler layer.  We don't have IVT to them.  We could make an extension at workspace layer, and update to that.  i empower you :)

Reply:
Isn't jt already expksed as an extension methkd by the compiler itself?

https://github.com/dotnet/roslyn/blob/2d7ee0d5da8eebaf39e6d01fb6be6f67c3a64d55/src/Compilers/CSharp/Portable/CSharpExtensions.cs#L144

Reply:
Good find